doc/man7/migration_guide.pod

```diff
@@ -0,0 +1,1814 @@
+
+=item SHA512_Init, SHA512_Update, SHA512_Final and SHA512_Transform
+
+See L</Deprecated low-level digest functions>.
+
+=item SRP_Calc_A, SRP_Calc_B, SRP_Calc_client_key, SRP_Calc_server_key,
+SRP_Calc_u, SRP_Calc_x, SRP_check_known_gN_param, SRP_create_verifier,
+SRP_create_verifier_BN, SRP_get_default_gN, SRP_user_pwd_free, SRP_user_pwd_new,
+SRP_user_pwd_set0_sv, SRP_user_pwd_set1_ids, SRP_user_pwd_set_gN,
+SRP_VBASE_add0_user, SRP_VBASE_free, SRP_VBASE_get1_by_user, SRP_VBASE_init,
+SRP_VBASE_new, SRP_Verify_A_mod_N, SRP_Verify_B_mod_N
+
+There are no replacements for the SRP functions.
+
+=item SSL_CTX_set_tmp_dh_callback, SSL_set_tmp_dh_callback,
+SSL_CTX_set_tmp_dh and SSL_set_tmp_dh.
+
+These are used to set the Diffie-Hellman (DH) parameters that are to be used by
+servers requiring ephemeral DH keys. Instead applications should consider using
+the built-in DH parameters that are available by calling L<SSL_CTX_set_dh_auto(3)>
+or L<SSL_set_dh_auto(3)>. If custom parameters are necessary then applications can
+use the alternative functions L<SSL_CTX_set0_tmp_dh_pkey(3)> and
+L<SSL_set0_tmp_dh_pkey(3)>. There is no direct replacement for the "callback"
+functions. The callback was originally useful in order to have different
+parameters for export and non-export ciphersuites. Export ciphersuites are no
+longer supported by OpenSSL. Use of the callback functions should be replaced
+by one of the other methods described above.
+
+=item SSL_CTX_set_tlsext_ticket_key_cb.
+
+Use the new L<SSL_CTX_set_tlsext_ticket_key_evp_cb(3)> function instead.
+
+=item WHIRLPOOL, WHIRLPOOL_Init, WHIRLPOOL_Update, WHIRLPOOL_Final and WHIRLPOOL_BitUpdate
+
+See L</Deprecated low-level digest functions>.
+The Whirlpool algorithm has been moved to the L<Legacy Provider|/Legacy Algorithms>.
+
+=item X509_certificate_type
+
+This was an undocumented function. Applications can use L<X509_get0_pubkey(3)>
+and L<X509_get0_signature(3)> instead.
+
+=item X509_http_nbio and X509_CRL_http_nbio
+
+Use L<X509_load_http(3)> and L<X509_CRL_load_http(3)> instead.
+
+=back
+
+=head2 Using the FIPS Module in applications
+
+See L<fips_module(7)> and L<OSSL_PROVIDER-FIPS(7)> for details.
+
+=head2 OpenSSL command line application changes
+
+=head3 New applications
+
+L<'kdf'|openssl-kdf(1)> uses the new L<EVP_KDF(3)> API.
+L<'mac'|openssl-mac(1)> uses the new L<EVP_MAC(3)> API.
+
+=head3 Added options
+
+'-provider_path' and '-provider' are available to all apps and can be used
+multiple times to load any providers, such as the 'legacy' provider or
+third party providers. If used then the 'default' provider would also need to be
+specified if required. The '-provider_path' must be specified before the
+'-provider' option. 
+
+'-providers' used by 'list', which will display the list of loaded providers,
+their names, version and status.  It optionally displays their gettable parameters.
+
+'-objects' used by 'list' which will display the list of built in objects,
```

Comment:
`list` picked up a pile more options.  I'm not sure it's necessary to spell them out, a reference to `openssl-list(1)` might be sufficient.  For example: kdf-algorithms.

Reply:
good idea.